service/productService.js

```diff
@@ -0,0 +1,84 @@
+const productModel = require('../model/productModel');
+
+const invaliDataError = { err: {
+  code: 'invalid_data',
+  message: 'Wrong id format',
+} };
+
+// Necessário para os testes. Verifica se o parâmetro passado é um hexadecimal de 24 dígitos.
+const standarizedId = /^[0-9a-fA-F]{24}$/;
+
+const getAll = async () => productModel.listAll();
+
+const getById = async (id) => {
+  let result = '';
+  // Se a string passada bate com o regex...
+  if (standarizedId.test(id)) { result = await productModel.selectById(id); }
+  return result || invaliDataError;
```

Comment:
Acredito que o código fique mais limpo se usarmos um ternário em `result`.
Além disso, evitamos um `if`, e o uso de mutabilidade

```suggestion
  const result = standarizedId.test(id) ? await productModel.selectById(id) : null;
  return result || invaliDataError;
```